### ISSUE TITLE
Creates pdf before fonts-style is fully applied

### ISSUE THREAD

**Issue by SdkCwss** [hardkoded/puppeteer-sharp#1510]
Hi
I'm trying to create a pdf from a html template using puppeteer-sharp - headless chrome.
This works fine for about half of the requests - the other half returns a pdf where the fonts are not applied to the page. 
(The requests are perfectly similar.. same html-template, css - all! - it is not consistent--sometimes two, three good pdfs - then one-two-three bad.. totally random!)

- The Fonts (Roboto Mono and Titillium Web) are installed locally on the hosting server. (c:\windows\fonts)
- Puppeteer-sharp vers. 2.0.3 (newest)

-the Html-body is quiet big, including multiple images - base64 encoded.. (dont know if this information is relevant - but rather mention it) 
I've tried several things and used a lot of time googling this issue... 
I've tried using page.EvaluateExpressionHandleAsync on promises - but does not seem to work. 

The code:
//---------------------------------------------------------------
  private async Task<byte[]> GeneratePdf(string bodyHtml, string headerHtml, string footerHtml, string cssStyle,
            PageSettings settings)
        {
            var browserFetcher = new BrowserFetcher(new BrowserFetcherOptions
            {
                Path = Path.Combine(AssemblyDirectory, "chromium")
            });

            await browserFetcher.DownloadAsync(BrowserFetcher.DefaultRevision);

            var launchOptions = new LaunchOptions
            {
                Headless = true,
                ExecutablePath = browserFetcher.RevisionInfo(BrowserFetcher.DefaultRevision).ExecutablePath
            };

            using (var browser = await Puppeteer.LaunchAsync(launchOptions))
            using (var page = await browser.NewPageAsync())
            {
                if(!string.IsNullOrWhiteSpace(cssStyle))
                    await page.AddStyleTagAsync(new AddTagOptions {Content = cssStyle});

                await page.SetContentAsync(bodyHtml);

                //Custom promise trying to postpone the creation of the pdf - not working
                await page.EvaluateExpressionHandleAsync("window.isPdfReady"); 
                //default promise trying (once again) to postpone the creation of the pdf until the page is completly loaded.
                await page.EvaluateExpressionHandleAsync("document.fonts.ready");

                return await page.PdfDataAsync(new PdfOptions
                {
                    DisplayHeaderFooter = !string.IsNullOrWhiteSpace(headerHtml + footerHtml),
                    HeaderTemplate = headerHtml,
                    FooterTemplate = footerHtml,
                    Format = PaperFormat.A4,
                    Landscape = UseLandscape(settings),
                    MarginOptions = BuildMarginOptions(settings),
                    PrintBackground = settings.PrintBackground
                });
            }
        }

Any pointers to how / why would be greatly appreciated!

**Comment by scotthellewell** [comment]
I'm not sure if I have the same issue as you or not.  However, I found that if I have a link to a font in my head section of my html that I get only the pictures on the page and not the text.  I found this happens when linking to the google fonts.  I also found that if the link is http instead of https, that it all works fine.  I found this issue on Windows and linux.  I'm unable to reproduce this issue on Mac.  It is unrelated to puppeteer sharp, because I can reproduce it rarely steadily by calling chrome directly from the command line.  It seems to happen at times about 20% of the time, and at other times close to 80% of the time. But adding this line

                html = html.Replace("https://fonts.googleapis.com", "http://fonts.googleapis.com");

It then renders properly.
